internal/command/test.go

```diff
@@ -1028,6 +1029,34 @@ func buildInputVariablesForTest(run *moduletest.Run, file *moduletest.File, conf
 	return backend.ParseVariableValues(variables, config.Module.Variables)
 }
 
+// getEvalContextFromStates constructs an hcl.EvalContext based on the provided map
+// of TestFileState instances. It extracts the relevant information from the
+// states to create a context suitable for HCL evaluation, including the output
+// values of modules.
+//
+// Parameters:
+//   - states: A map of TestFileState instances containing the state information.
+//
+// Returns:
+//   - *hcl.EvalContext: The constructed HCL evaluation context.
+func getEvalContextFromStates(states map[string]*TestFileState) *hcl.EvalContext {
+	runCtx := make(map[string]cty.Value)
+	for _, state := range states {
+		if state.Run == nil {
+			continue
+		}
+		outputs := make(map[string]cty.Value)
+		mod := state.State.Modules[""] // Empty string is what is used by the module in the test runner
+		for outName, out := range mod.OutputValues {
+			outputs[outName] = out.Value
+		}
+		runCtx[state.Run.Name] = cty.ObjectVal(outputs)
+	}
+	ctx := &hcl.EvalContext{Variables: map[string]cty.Value{"run": cty.ObjectVal(runCtx)}}
+
+	return ctx
+}
```

Comment:
Am I correct in thinking that this logic is copied from [here](https://github.com/opentofu/opentofu/pull/1129/files#diff-16f7ccd0b34c619fda941531b649e919dcc79d87306854db9d212c3050296305R1062-R1076) ?

If so I think we could look at abstracting that logic out to a common method and updating the logic in command/test.go

Reply:
Done